### PR TITLE
Add draft for CRUD based gremlin design.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
## Description
Add draft for CRUD based gremlin design.

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos
- [ ] Tests
- [ ] Documentation

## Steps to Test
Steps to test code change

### CODE REVIEW

File: DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Spring Data Gremlin Design
+
+### Orientation
+
+Gremlin is a functional, data-flow language that enables users to succinctly express complex 
+traversals on (or queries of) their application's property graph. It hides the details of backend
+database implementation (like azure cosmosdb support Graph API).
+
+Apache Tinkerpop gremlin java driver allows you to launch gremlin query, but it is not easy to
+get familiar with gremlin syntax. You must to generate all gremlin queries by your self as following.
```
Comment (yaweiw) [inline DESIGN.md]: must to -> have to

File: DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        "g.addV('person').property('id', '1').property('name', 'pli').property('age', 31)",
+        "g.addV('person').property('id', '4').property('name', 'incarnation').property('age', 27)",
+        "g.addV('software').property('id', '2').property('name', 'spring-boot-sample').property('lang', 'java')",
+        "g.V('1').addE('created').to(g.V('2')).property('weight', 0.8)",
+        "g.V('1').addE('contributed').to(g.V('2')).property('weight', 0.1)",
+        "g.V('4').addE('contributed').to(g.V('2')).property('weight', 0.4)"
+    };
+```
+
+We would like to make things better, and try to hide the process of mapping Java instance to graph
```
Comment (yaweiw) [inline DESIGN.md]: better -> easier

File: DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+database persistent entity, with the help of spring-data-commons.
+
+### From Users' View
+How a normal user leverage graph database ? Basically, it can use gremlin driver to generate the
+query literal with you java class instance. But like almost all database in spring, we help users
```
Comment (yaweiw) [inline DESIGN.md]: How do users use graph db? They can use gremlin driver to generate the query literal in Java class instance. It's not impossible but yet isn't easy. We want to figure out a more Spring natural way by leveraging Spring annotations to map Java instance to database entity.

File: DESIGN.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+With the above constriction from gremlin, `GremlinSource` has field `id` and `label` for mapping,
+and keep all other fields into one `Map<String, Object>`. When we try to insert a instance from
+java to `GremlinSource`, we perform a **WRITE** operation and convert the instance to a `GremlinSource`.
+Operation **WRITE** will convert all the data of one instance to `GremlinSource`, and take care of 
+`id` or `@Id` field.
+
+The `GremlinSourceWriter` converts Java instance to `GremlinSource`.
+
+##### GremlinScript
```
Comment (Matthew-Dong) [inline DESIGN.md]: Suggest put this session after GremlinResult

**Pull request merged**